Extensions/DialogueTree/dialoguetools.js

```diff
@@ -0,0 +1,538 @@
+ */
+gdjs.dialogueTree.loadFromSceneVar = function(
+  runtimeScene,
+  sceneVar,
+  startDialogueNode
+) {
+  this.runner = gdjs.dialogueTree.runner;
+  this.yarnData = JSON.parse(sceneVar.getAsString());
+  this.runner.load(this.yarnData);
+
+  if (startDialogueNode && startDialogueNode.length > 0) {
+    gdjs.dialogueTree.startFrom(startDialogueNode);
+  }
+};
+
+/**
+ * Load the Dialogue Tree data of the game. Use this action to initialize The Dialogue Tree, so as it can be used in the game
+ * @param {gdjs.Variable} jsonResourceName The json fileresource to load the Dialogue tree data from. The data is a JSON file - created by YARN
+ * @param {string} startDialogueNode The Dialogue Branch to start the Dialogue Tree from. If left empty, the data will only be loaded, but can later be initialized via another action
+ */
+gdjs.dialogueTree.loadFromJsonFile = function(
+  runtimeScene,
+  jsonResourceName,
+  startDialogueNode
+) {
+  runtimeScene
+    .getGame()
+    .getJsonManager()
+    .loadJson(jsonResourceName, function(error, content) {
+      if (error) {
+        console.error("An error happened:", error);
+      } else {
+        if (!content) return;
+        gdjs.dialogueTree.yarnData = content;
+        gdjs.dialogueTree.runner.load(gdjs.dialogueTree.yarnData);
+
+        if (startDialogueNode && startDialogueNode.length > 0) {
+          gdjs.dialogueTree.startFrom(startDialogueNode);
+        }
+      }
+    });
+};
+
+// Condition to check if the Dialogue Tree is currentlyu parsing data. Use this to do things like disabling player movement while talking on an npc
+gdjs.dialogueTree.isRunning = function() {
+  return this.dialogueIsRunning;
+};
+
+// Action to scroll the clipped text. Use this with a timer and user input to control how fast dialogue line text is scrolling.
+gdjs.dialogueTree.scrollCippedText = function() {
+  if (this.pauseScrolling || !this.dialogueIsRunning) return;
+
+  if (this.dialogueText) {
+    this.clipTextEnd += 1;
+  }
+};
+
+// Condition to check if text scrolling has completed.
+// Use this action to prevent the user from skipping to next line before the current one has been printed fully
+gdjs.dialogueTree.cippedTextScrollingHasCompleted = function() {
+  if (this.dialogueData && this.dialogueText.length) {
+    return this.clipTextEnd >= this.dialogueText.length;
+  }
+  return false;
+};
+
+// Expression to get the current dialogue line with a scrolling effect (recommended)
+// Use this with the scrollCippedText action to achieve a classing scrolling text,as well as any <<wait>> effects to pause scrolling
+gdjs.dialogueTree.getClippedLineText = function() {
+  return this.dialogueText.length
+    ? this.dialogueText.substring(0, this.clipTextEnd)
+    : "";
+};
+
+// Expression to get the current complete dialogue line without using any scrolling effects.
```

Comment:
I let you remove expression/action/condition and rework the comments for the rest of the file.